pkg/ring/consul_client.go

```diff
@@ -110,7 +113,9 @@ func (c *consulClient) cas(ctx context.Context, key string, f CASCallback) error
 		if kvp != nil {
 			out, err := c.codec.Decode(kvp.Value)
 			if err != nil {
-				level.Error(util.Logger).Log("msg", "error decoding key", "key", key, "err", err)
+				if resp, ok := httpgrpc.HTTPResponseFromError(err); ok && resp.GetCode() != 202 {
+					level.Error(util.Logger).Log("msg", "error decoding key", "key", key, "err", err)
+				}
```

Comment:
I don't see how this if will ever not be true - none of the codecs return a httpgrpc error.

Reply:
Yeah I think I accidentally added this here but meant to add the check in the http code, which I did 5 minutes later. I'll remove this and deploy all the changes from the review. If there isn't logging here for 202's then I'd mistakenly added this check :)